spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/ConversionUtils.java

```diff
@@ -54,4 +60,46 @@ public static boolean isIterableNonByteArrayType(Class propType) {
 		source.forEach((item) -> result.add(converter.convert(item, targetType)));
 		return result;
 	}
+
+	public static <T> T wrapSimpleLazyProxy(Supplier<T> supplierFunc, Class<T> type) {
+		return (T) Proxy.newProxyInstance(type.getClassLoader(), new Class[] { type },
+				new SimpleLazyDynamicInvocationHandler<>(supplierFunc));
+	}
+
+	public static boolean isUnevaluatedLazyProxy(Object object) {
```

Comment:
The result is negated the only time this method is used. Can we make it `isEvaluatedLazyProxy` and avoid negation?

Reply:
I had that originally but that is decidedly less useful. 

assuming we use "isEvaluatedLazyProxy" this would evaluate to FALSE if the property in question is not a proxy at all. 

instead, "isUnevaluatedLazyProxy" evaluating to true represents all the cases we DON'T want to save-entity, because it means it is NOT a real object and also NOT an evaluated-proxy

Reply:
I think double negation here looks confusing: `!ConversionUtils.isUnevaluatedLazyProxy(kids)` 
Maybe rename the method?

Reply:
renamed.